src/main/java/control/Pulsador.java

```diff
@@ -1,65 +1,65 @@
-package control;
-
-import dominio.Personaje;
-import dominio.Tablero;
-import dominio.Tablero.Estado;
-
-public class Pulsador implements Runnable {
-
-	Tablero tablero;
-	private int tecla;
-	public boolean modificado;
-
-	public Pulsador(Tablero tablero) {
-		this.tablero = tablero;
-	}
-
-	synchronized public void run() {
-		Estado estado = Estado.QUIETO;
-		while (true) {
-			
-			try {
-				if (modificado && 	!((Personaje)(tablero.getElementos().get(0))).isPausado()  ) {
-					Thread.sleep(50);
-					estado =  tablero.moverPersonaje(tecla);
-					Control.getInstance().getVista().cambioDeImagen(estado, Control.getInstance().getVista().getMario());
-					Control.getInstance().getVista().cambioPixeles(estado);
-					modificado = false;
-				}else{
-				Thread.sleep(50);
-				}
-			} catch (Exception e) {
-			}
-	
-		}
-	}
-	
-	
-	
-
-	public Tablero getTablero() {
-		return tablero;
-	}
-
-	public void setTablero(Tablero tablero) {
-		this.tablero = tablero;
-	}
-
-	public int getTecla() {
-		return tecla;
-	}
-
-	public void setTecla(int tecla) {
-		this.modificado = true;
-		this.tecla = tecla;
-	}
-
-	public boolean isModificado() {
-		return modificado;
-	}
-
-	public void setModificado(boolean modificado) {
-		this.modificado = modificado;
-	}
-
-}
+package control;
+
+import dominio.Personaje;
+import dominio.Tablero;
+import dominio.Tablero.Estado;
+
+public class Pulsador implements Runnable {
+
+	Tablero tablero;
+	private int tecla;
+	public boolean modificado;
+
+	public Pulsador(Tablero tablero) {
+		this.tablero = tablero;
+	}
+
+	synchronized public void run() {
+		Estado estado = Estado.QUIETO;
+		while (true) {
+			
+			try {
+				if (modificado && 	!((Personaje)(tablero.getElementos().get(0))).isPausado()  ) {
+					Thread.sleep(50);
+					estado =  tablero.moverPersonaje(tecla);
+					Control.getInstance().getVista().cambioDeImagen(estado, Control.getInstance().getVista().getMario());
+					Control.getInstance().getVista().cambioPixeles(estado);
+					modificado = false;
+				}else{
+				Thread.sleep(50);
+				}
+			} catch (Exception e) {
+			}
+	
+		}
+	}
+	
+	
+	
+
+	public Tablero getTablero() {
+		return tablero;
+	}
+
+	public void setTablero(Tablero tablero) {
+		this.tablero = tablero;
+	}
+
+	public int getTecla() {
+		return tecla;
+	}
+
+	public void setTecla(int tecla) {
```

Comment:
estáis mezclando el POJO (la representación de Pulsador) con su controlador y os lleva a cosas cómo cambiar el valor de dos atributos en el set de uno.
El run también se movería a PulsadorController
Pulsador y PulsadorController dónde en PulsadorController.press(tecla: int) modifica el valor de 
```
pulsador.setTecla(tecla);
pulsador.setModificado(true);
```

Parecen tonterías, pero esto solo genera poltergeists

Reply:
Hay una máxima en la programación "respetemos los POJOS" solo representan objetos, constructor, atributos y sus getters y setters y esto vale para software de gestión cómo para mandar cohetes a la luna